docs/specification/draft/basic/transports.md

```diff
@@ -48,38 +48,197 @@ sequenceDiagram
+   - If the server initiates an SSE stream:
+     - The server **MAY** close the SSE stream at any time.
+     - The client **MAY** close the SSE stream at any time.
+
+### Session Management
+
+Where a client desires to share a single logical session across multiple requests, it
+**MAY** attach an `Mcp-Session-Id` HTTP header to its requests. This permits maintenance
+of session state across separate POSTs.
+
+1. It is the client's responsibility to generate or select the session ID.
+2. This session ID **SHOULD** be globally unique and cryptographically secure (e.g., a
+   UUID or a JWT), unless it is specifically desired to share a session ID across users
+   or clients.
+3. The server **MAY** use this header to associate state with the logical session.
+   - If the server agrees to re-establish an existing session, it **MUST** include a
+     `Mcp-Session-Status: resumed` HTTP header in its response.
+   - If the server begins a new session associated with the session ID, it **MUST**
+     include a `Mcp-Session-Status: created` HTTP header in its response.
+   - The server **MAY** delete session state at any time; however, it **MUST NOT**
+     prevent the same session ID from creating a new session again in future.
+4. If [authorization]({{< ref "authorization" >}}) is used _and_ the server makes use of
+   the `Mcp-Session-Id` header:
+   - The server **SHOULD** bind the session ID to the authorization context, and return
+     an error if the session ID is reused in a different authorization context.
+
+### Multiple Connections
+
+1. The client **MAY** remain connected to multiple SSE streams simultaneously.
+2. The server **MUST** send each of its JSON-RPC messages on only one of the connected
+   streams; that is, it **MUST NOT** broadcast the same message across multiple streams.
+   The server **MAY** use different streams for _different_ messages.
+   - The server can mitigate the risk of message loss by supporting
+     [resumability and redelivery](#resumability-and-redelivery).
+
+### Resumability and Redelivery
+
+To support resuming broken connections, and redelivering messages that might otherwise be
+lost:
+
+1. Servers **MAY** attach an `id` field to their SSE events, as described in the
+   [SSE standard](https://html.spec.whatwg.org/multipage/server-sent-events.html#event-stream-interpretation).
+   - If present, the ID **MUST** be globally unique across all streams within that
+     [session](#session-management)—or all streams with that specific client, if session
+     management is not in use.
+2. After a broken connection, clients **MAY** include a
+   [`Last-Event-ID`](https://html.spec.whatwg.org/multipage/server-sent-events.html#the-last-event-id-header)
+   header when opening a new SSE stream, to indicate the last event ID they received.
+   - The server **MAY** use this header to replay messages that were sent after the last
+     event ID, and to resume the stream from that point.
```

Comment:
Iis there also the opportunity here to use similar semantics for a JSON-RPC only protocol? 

eg  Client issues `GET` to `/mcp` with the `application/json` accept type, and the `last-event-id` header. The server could respond with a JSON array containing all the missed messages since that id. 

Might be a slightly crazy idea - but thought I'd throw it out there. 🙃

Reply:
Seems better to do this via SSE instead, since `last-event-id` is part of that spec. We can explore this as an extension in future if needed.